apps/extension/server/src/server.ts

```diff
@@ -565,12 +559,39 @@ connection.onRequest('getMachineAtCursorPosition', ({ uri, position }) => {
   };
 });
 
+connection.onRequest('applyMachineEdits', ({ machineEdits }) => {
+  if (!displayedMachine) {
+    throw new Error(
+      '`applyMachineEdits` can only be requested when there is a displayed machine',
+    );
+  }
+
+  const modified = documentsCache
+    .get(displayedMachine.uri)!
```

Comment:
Nit: can we guarantee this always exists?

Reply:
Yes, we can't have a displayed machine without having a cached document for it.